colmena/proxy.py

```diff
@@ -70,9 +70,18 @@ def resolve_proxies_async(args: Union[object, list, tuple, dict]) -> None:
     Args:
         args (object, list, tuple, dict): possible object or
             iterable of objects that may be ObjectProxy instances
+
+    Returns:
+        List of the proxies that are being resolved
     """
-    def resolve_async_if_proxy(obj: Any) -> None:
+
+    # Create a list to store the keys
+    output = []
+
+    # Make a function that will resolve proxies
+    def resolve_async_if_proxy(obj: Any) -> NoReturn:
```

Comment:
This function returns `None` implicitly. `NoReturn` is used for function that never returns control flow to a caller. For example, a `while True` loop that never terminates or a function that raises an exception.

Reply:
That's very good to know 🙇‍♂️